_pytest/mark.py

```diff
@@ -94,6 +91,20 @@ def _for_parameterize(cls, argnames, argvalues, function):
         return argnames, parameters
 
 
+def get_empty_parameterset_mark(config, argnames, function):
+    requested_mark = config.getini('empty_parameterset')
+    if requested_mark in ('', None, 'skip'):
+        mark = MARK_GEN.skip
+    elif requested_mark == 'xfail':
+        mark = MARK_GEN.xfail(run=False)
+    else:
+        raise LookupError(requested_mark)
+    fs, lineno = getfslineno(function)
+    reason = "got empty parameter set %r, function %s at %s:%d" % (
+        argnames, function.__name__, fs, lineno)
+    return mark(reason=reason)
```

Comment:
For `requested_mark == 'xfail'` this will be equivalent to `MARK_GEN.xfail(run=False)(reason=reason)`... is this right?

Reply:
correct

Reply:
Weird, so `pytest.mark.xfail(run=False)(reason=reason)` produces the correct result?

Reply:
@nicoddemus pytest marks support compounding addition of args/kwargs

`mark.foo(1)(2)(2)(b=2) == mark.foo(1,2,3,b=3)`